openpype/hosts/houdini/api/lib.py

```diff
@@ -558,19 +558,33 @@ def get_frame_data(node):
         dict: frame data for star, end and steps.
 
     """
+    if asset_data is None:
+        asset_data = {}
+
     data = {}
 
     if node.parm("trange") is None:
-
+        self.log.debug(
+            "Node has no 'trange' parameter: {}".format(node.path())
+        )
         return data
 
     if node.evalParm("trange") == 0:
-        self.log.debug("trange is 0")
+        self.log.debug(
+            "Node '{}' has 'Render current frame' set. "
+            "Time range data ignored.".format(node.path())
+        )
         return data
 
-    data["frameStart"] = node.evalParm("f1")
-    data["frameEnd"] = node.evalParm("f2")
-    data["steps"] = node.evalParm("f3")
+    data["frameStartHandle"] = node.evalParm("f1")
+    data["frameStart"] = node.evalParm("f1") + asset_data.get("handleStart", 0)
+    data["handleStart"] = asset_data.get("handleStart", 0)
+
+    data["frameEndHandle"] = node.evalParm("f2")
+    data["frameEnd"] = node.evalParm("f2") - asset_data.get("handleEnd", 0)
+    data["handleEnd"] = asset_data.get("handleEnd", 0)
```

Comment:
```suggestion
    data["frameStartHandle"] = node.evalParm("f1")
    data["handleStart"] = asset_data.get("handleStart", 0)
    data["frameStart"] = data["frameStartHandle"] + data"handleStart"]

    data["frameEndHandle"] = node.evalParm("f2")
    data["handleEnd"] = asset_data.get("handleEnd", 0)
    data["frameEnd"] =  data["frameEndHandle"] - data["handleEnd"]
```

Thanks for keeping the ROP node values as the ground truth for the output range instead of **adding** the handle frames on top of the data. ❤️ 

It might happen however that someone would export a frame range of say 1001-1002 or even 1001-1001 and that the asset handles are start handles 5 and end handles 5 which makes `frameStart` 1006 and `frameEnd` 996 - basically an invalid range.

Due to this complexity now we might need to add a validator for those cases - however, it'd mean that if the asset's handles are configured to be five start and end handles you'd never be able as an artist to e.g. export two frames.

Reply:
>  basically an invalid range.

I spotted that problem while testing and ...
![image](https://github.com/ynput/OpenPype/assets/20871534/c013af20-b7bf-4d85-b6a0-ec674468373d)

> This might be quite an awkward extra complexity we're now adding to the logic?

what about a simple `if frameEnd < frameStart ` ?

Reply:
Well, validating is easy - but _what do we want to do_ if all the artist really wants is to just output three frames, no more, no less when the artist knows what he is doing. We don't want to disallow that.